aws/amazon-chime-sdk-js
# Add a mechanism to clear the deviceInfoCache

**Issue #1305 :**

**Description of changes:**
Add an optional parameter `force` in `listAudioInputDevices`, `listAudioOutputDevices`, `listVideoInputDevices` and `listDevicesOfKind`. User can force update the `deviceInfoCache` from browser with parameter `force` set to `true`.

**Testing**

1. Have you successfully run `npm run build:release` locally?
Yes
2. How did you test these changes?
Write a unit test to cover the change.
* call `listAudioInputDevices` to establish `deviceInfoCache`
* check `updateDeviceInfoCacheFromBrowser` is called once
* call `listAudioInputDevices` with `force` parameter set to `true`
* check `updateDeviceInfoCacheFromBrowser` is called twice

3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
No
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
Yes
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
